compile.bash

```diff
@@ -10,13 +10,16 @@ mkdir build
 cd build
 
 cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_SYSCONFDIR=/etc \
-	-DCMAKE_INSTALL_LOCALSTATEDIR=/var -DICINGA2_RUNDIR=/run \
+	-DCMAKE_INSTALL_LOCALSTATEDIR=/var -DICINGA2_RUNDIR=/run -DICINGA2_LTO_BUILD=ON \
 	-DICINGA2_SYSCONFIGFILE=/etc/sysconfig/icinga2 -DICINGA2_WITH_{COMPAT,LIVESTATUS}=OFF ..
 
 make
 make test
 
 make install "DESTDIR=$(pwd)/../icinga2-bin"
+strip -g ../icinga2-bin/usr/lib/*/icinga2/sbin/icinga2
+strip -g ../icinga2-bin/usr/lib/nagios/plugins/check_nscp_api
```

Comment:
I find this change surprising. I'd expect this to happen with `CMAKE_BUILD_TYPE=Release` but apparently that's not the case. I've dug around a bit in the git history and found the following:

- https://github.com/Icinga/icinga2/commit/e5fba1d881c7de0b16ddf8fd662edb9699cc8b44 + https://github.com/Icinga/icinga2/issues/958
- https://github.com/Icinga/icinga2/commit/b6220c6ecfd3a64bf73ae5de8ed62f257867b5c0 + https://github.com/Icinga/icinga2/issues/998

So apparently someone felt the need that release builds need debug symbols, but it remains unclear why.

Reply:
And... ?

Reply:
You want to do a change, so it's up to you to provide the reasoning why this change is a good idea.

Reply:
Well, because CMAKE_BUILD_TYPE=Release doesn’t do what we want? 🤷‍♂️

Reply:
* [icinga2/CMakeLists.txt](https://github.com/Icinga/icinga2/blob/4b754637f7ac06328d101c72fd03e897707fd352/CMakeLists.txt#L251-L265): "All builds shall have debug symbols."
* This PR: "I don't think so!"

So far, you didn't provide reasoning why stripping the debug symbols is a good idea. For example, may this change affect the crash log?

Reply:
All our _builds_, yes. But not all our _installs_. I mean, our package installs come w/o debug info by default. For this there are extra packages which indeed need it. Do you wanna carry those additional 228MB in every Docker install and so nearly double the image size?

Reply:
Good point, the regular packages should basically do the same, so debug symbols can't be very important for normal operation. So the only practical implication might be that you can't easily get debug symbols for the container images. (Is there a "Docker way of doing" for this? Would you typically provide a second image for this?)

But still strange to me why the `CMakeLists.txt` is doing that. I don't get it, isn't that precisely the reason why `Release` and `RelWithDebInfo` exist? Do we also want to do something about that over there?

Reply:
1. Let’s wait until (unlikely) someone actually complains.
2. Let’s solve one problem at a time.

Reply:
> Let’s solve one problem at a time.

https://github.com/Icinga/icinga2/issues/9623